cfme/configure/tasks.py

```diff
@@ -70,39 +167,27 @@ def is_task_finished(destination, task_name, expected_status, clear_tasks_after_
     return True
 
 
-def is_analysis_finished(name, task_type='vm', clear_tasks_after_success=True):
-    """ Check if analysis is finished - if not, reload page"""
+def wait_analysis_finished_multiple_tasks(
+        task_name, task_type, expected_num_of_tasks, delay=5, timeout='5M'):
+    """ Wait until analysis is finished (or timeout exceeded)"""
+    row_completed = []
+    # get view for reload button
+    view = navigate_to(Tasks, 'AllTasks')
 
-    tabs_data = {
-        'container': {
-            'tab': 'AllTasks',
-            'task': '{}',
-            'state': 'finished'
-        },
-        'vm': {
-            'tab': 'AllTasks',
-            'task': 'Scan from Vm {}',
-            'state': 'finished'
-        },
-        'host': {
-            'tab': 'MyOtherTasks',
-            'task': "SmartState Analysis for '{}'",
-            'state': 'Finished'
-        },
-        'datastore': {
-            'tab': 'MyOtherTasks',
-            'task': 'SmartState Analysis for \[{}\]',
-            'state': "Finished"
-        },
-        'cluster': {
-            'tab': 'MyOtherTasks',
-            'task': 'SmartState Analysis for \[{}\]',
-            'state': "Finished"}
-    }[task_type]
-    return is_task_finished(destination=tabs_data['tab'],
-                            task_name=tabs_data['task'].format(name),
-                            expected_status=tabs_data['state'],
-                            clear_tasks_after_success=clear_tasks_after_success)
+    def tasks_finished(output_rows=row_completed, task_name=task_name,
```

Comment:
**Required**

The default for a method should not be mutable, in this case an empty list.
http://docs.python-guide.org/en/latest/writing/gotchas/

Reply:
Mike I have added this tasks_finished since i wanted to avoid return of tuple , when calling this method with wait_for.
Since i shouldn't use mutable value as default for this method,
would it be right that i will just call are_all_tasks_match_status method (instead of tasks_finished method) like the following:
    try:
        wait_for_result, duration = wait_for(are_all_tasks_match_status,
                                                               func_kwargs={
                                                                      'task_name': task_name,
                                                                      'expected_num_of_tasks': expected_num_of_tasks,
                                                                      'task_type': task_type},
                                                               delay=delay,
                                                               timeout=timeout,
                                                               fail_func=view.reload.click)
        return wait_for_result[1]

Reply:
Remove the default for this method.
I pass value as func_kwargs when i call this method with wait_for.